cdi: Renaming binaries in the SPEC file to align with upstream naming conventions

sharath-srikanth-chellappa opened the pull request:
<!--
COMMENT BLOCKS WILL NOT BE INCLUDED IN THE PR.
Feel free to delete sections of the template which do not apply to your PR, or add additional details
-->

The cdi binaries are misnamed in the SPEC file leading to additional Dockerfile commands to rename the binaries. Changing the SPEC file naming conventions to align with upstream and removing Dockerfile workarounds.


###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of the changes. -->
What does the PR accomplish, why was it needed?

The cdi binaries are misnamed in the SPEC file leading to additional Dockerfile commands to rename the binaries. Changing the SPEC file naming conventions to align with upstream and removing Dockerfile workarounds.

###### Change Log  <!-- REQUIRED -->
<!-- Detail the changes made here. -->
<!-- Please list any packages which will be affected by this change, if applicable. -->
<!-- Please list any CVES fixed by this change, if applicable. -->


###### Does this affect the toolchain?  <!-- REQUIRED -->
<!-- Any packages which are included in the toolchain should be carefully considered. Make sure the toolchain builds with these changes if so. -->
<!-- Update: manifests/package/toolchain_*.txt, pkggen_core_*.txt, update_manifests.sh -->
<!-- To validate: make clean; make workplan REBUILD_TOOLCHAIN=y DISABLE_UPSTREAM_REPOS=y CONFIG_FILE="" ... -->
**NO**

###### Associated issues  <!-- optional -->
<!-- Link to Github issues if possible. -->
<!-- you can use "fixes #xxxx" to auto close an associated issue once the PR is merged -->
- #56050776

###### Links to CVEs  <!-- optional -->


###### Test Methodology
<!-- How was this test validated? i.e. local build, pipeline build etc. -->
- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=727053&view=results
- Container Pipeline Build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=728225&view=results

```
(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-uploadserver:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet
[sudo] password for sharath:

acrafoimages.azurecr.io/base/kubevirt/cdi-uploadserver:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)

(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-uploadproxy:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet

acrafoimages.azurecr.io/base/kubevirt/cdi-uploadproxy:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)

(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-operator:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet

acrafoimages.azurecr.io/base/kubevirt/cdi-operator:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)

(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-importer:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet

acrafoimages.azurecr.io/base/kubevirt/cdi-importer:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)

(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-controller:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet

acrafoimages.azurecr.io/base/kubevirt/cdi-controller:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)

(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-cloner:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet

acrafoimages.azurecr.io/base/kubevirt/cdi-cloner:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)

(base) sharath@sharath-2:~$ sudo trivy image  --exit-code 1 --severity HIGH,CRITICAL,MEDIUM "acrafoimages.azurecr.io/base/kubevirt/cdi-apiserver:1.57.0-10.3.0.20250204-amd64" --ignore-unfixed --quiet

acrafoimages.azurecr.io/base/kubevirt/cdi-apiserver:1.57.0-10.3.0.20250204-amd64 (azurelinux 3.0)

Total: 0 (MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

romoh commented:
LGTM

mandeepsplaha reviewed (commented):
The changes look good. Can you please run a container build for the cdi components and ensure they build successfully, and can you also link the build run in the PR description?

sharath-srikanth-chellappa commented:
> The changes look good. Can you please run a container build for the cdi components and ensure they build successfully, and can you also link the build run in the PR description?

Added that information to the PR description. Also added a manual trivy scan of the container images
